File: trunk/apps/sequencers/midibox_seq_v4/macos/Classes/Tasks.m
//
//  Tasks.m
//  midibox_seq_v4
//
//  Created by Thorsten Klose on 30.06.09.
//  Copyright 2009 __MyCompanyName__. All rights reserved.
//

#import "Tasks.h"

#include <mios32.h>


@implementation Tasks

// local variables to bridge objects to C functions
static NSObject *_self;

// semaphores used by ../core/tasks.h
NSRecursiveLock *SDCardSemaphore;
NSRecursiveLock *MIDIINSemaphore;
NSRecursiveLock *MIDIOUTSemaphore;
NSRecursiveLock *LCDSemaphore;


//////////////////////////////////////////////////////////////////////////////
// init local variables
//////////////////////////////////////////////////////////////////////////////
- (void) awakeFromNib
{
	_self = self;
	
	SDCardSemaphore = [[NSRecursiveLock alloc] init];
	MIDIINSemaphore = [[NSRecursiveLock alloc] init];
	MIDIOUTSemaphore = [[NSRecursiveLock alloc] init];
	LCDSemaphore = [[NSRecursiveLock alloc] init];
}


//////////////////////////////////////////////////////////////////////////////
// Semaphore access functions for C code
//////////////////////////////////////////////////////////////////////////////

void TASKS_SDCardSemaphoreTake(void) { [SDCardSemaphore lock]; }
void TASKS_SDCardSemaphoreGive(void) { [SDCardSemaphore unlock]; }

void TASKS_MIDIINSemaphoreTake(void) { [MIDIINSemaphore lock]; }
void TASKS_MIDIINSemaphoreGive(void) { [MIDIINSemaphore unlock]; }

void TASKS_MIDIOUTSemaphoreTake(void) { [MIDIOUTSemaphore lock]; }
void TASKS_MIDIOUTSemaphoreGive(void) { [MIDIOUTSemaphore unlock]; }

void TASKS_LCDSemaphoreTake(void) { [LCDSemaphore lock]; }
void TASKS_LCDSemaphoreGive(void) { [LCDSemaphore unlock]; }

@end

File: trunk/apps/synthesizers/midibox_sid_v3/juce/Source/tasks.c
// $Id$
/*
 * Temporary Wrapper code/stubs for Application Tasks
 * Only functions used by MBSID applications are listed here!
 * TODO: emulate FreeRTOS in background via Posix driver
 *
 * ==========================================================================
 *
 *  Copyright (C) 2009 Thorsten Klose (tk@midibox.org)
 *  Licensed for personal non-commercial use only.
 *  All other rights reserved.
 * 
 * ==========================================================================
 */

#include <mios32.h>

s32 TASKS_SDCardSemaphoreTake(void) { return -1; }
s32 TASKS_SDCardSemaphoreGive(void) { return -1; }
s32 TASKS_MIDIINSemaphoreTake(void) { return -1; }
s32 TASKS_MIDIINSemaphoreGive(void) { return -1; }
s32 TASKS_MIDIOUTSemaphoreTake(void) { return -1; }
s32 TASKS_MIDIOUTSemaphoreGive(void) { return -1; }
s32 TASKS_LCDSemaphoreTake(void) { return -1; }
s32 TASKS_LCDSemaphoreGive(void) { return -1; }

s32 TASKS_Init(u32 mode) { return -1; }

s32 SEQ_TASK_SID(void) { return -1; }
s32 SEQ_TASK_Period1mS(void) { return -1; }
s32 SEQ_TASK_Period1mS_LowPrio(void) { return -1; }
s32 SEQ_TASK_Period1S(void) { return -1; }

s32 TASK_MSD_EnableSet(u8 enable) { return -1; }
s32 TASK_MSD_EnableGet() { return -1; }
s32 TASK_MSD_FlagStrGet(char str[5]) { return -1; }
